cli/src/fetch-recipe.ts

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import fetch, { Response } from 'node-fetch';
+import { promises as fs } from 'fs';
+
+import { Recipe } from './recipe';
+import StatusCodes from './status-codes';
+
+type RecipeFile = {
+  type: 'file';
+  name: string;
+  download_url: string;
+};
+
+type RecipeDir = {
+  type: 'dir';
+  name: string;
+  path: string;
+};
+
+type RecipeNode = RecipeFile | RecipeDir;
+
+const COOKBOOK_BASE_URL =
+  'https://api.github.com/repos/Wolox/frontend-cookbook/contents';
+
+const COOKBOOK_FOLDERS_EXCEPTIONS: Record<string, string> = {
+  react: 'cookbook-react/src/recipes'
+};
+const getCookbookUrl = (tech: string) =>
+  COOKBOOK_FOLDERS_EXCEPTIONS[tech] || `cookbook-${tech}/recipes`;
+
+const handleResponseError = (
+  response: Response,
+  errorMessages: () => { notFoundError?: string; clientError?: string }
```

Comment:
Why is this a function? I feel like it could just be an object

Reply:
Because I don't want to create the object unless there is an error.
that's the way I'm avoiding the double `response.json()` that crashes